src/libraries/System.Text.Json/tests/Serialization/PropertyVisibilityTests.NonPublicAccessors.cs

```diff
@@ -0,0 +1,332 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static partial class PropertyVisibilityTests
```

Comment:
Currently this PR assumes that the property is still `public`, correct?(meaning the setter or getter can be non-public, but not both). Do we have tests with a private property where both setter and getter are private to verify?

Reply:
Yes, I added some tests showing that non-public properties (with various accessor visibility levels) are not included: https://github.com/dotnet/runtime/blob/e4e05858e9689099b907387a1159de6a997d65d7/src/libraries/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs#L869-L891

These tests ensure that we throw `IOE` when `[JsonInclude]` is used on a non-public properties: https://github.com/dotnet/runtime/blob/e4e05858e9689099b907387a1159de6a997d65d7/src/libraries/System.Text.Json/tests/Serialization/PropertyVisibilityTests.NonPublicAccessors.cs#L295-L330